git_machete/cmd.py

```diff
@@ -196,145 +57,19 @@ def pick(choices: List[str], name: str, apply_fmt: bool = True) -> str:
     return choices[index]
 
 
-def debug(cli_ctxt: CommandLineContext, hdr: str, msg: str) -> None:
-    if cli_ctxt.opt_debug:
-        sys.stderr.write(f"{bold(hdr)}: {dim(msg)}\n")
-
-
 # To avoid displaying the same warning multiple times during a single run.
 displayed_warnings: Set[str] = set()
 
 
 def warn(msg: str, apply_fmt: bool = True) -> None:
-    global displayed_warnings
     if msg not in displayed_warnings:
         sys.stderr.write(colored("Warn: ", RED) + (fmt(msg) if apply_fmt else msg) + "\n")
         displayed_warnings.add(msg)
 
 
-def directory_exists(path: str) -> bool:
-    try:
-        # Note that os.path.isdir itself (without os.path.abspath) isn't reliable
-        # since it returns a false positive (True) for the current directory when if it doesn't exist
-        return os.path.isdir(os.path.abspath(path))
-    except OSError:
-        return False
-
-
-def current_directory_or_none() -> Optional[str]:
-    try:
-        return os.getcwd()
-    except OSError:
-        # This happens when current directory does not exist (typically: has been deleted)
-        return None
-
-
 # Let's keep the flag to avoid checking for current directory's existence
```

Comment:
This comment applies to `current_directory_confirmed_to_exist`, should be moved together